docs/spanner.md

```diff
@@ -0,0 +1,189 @@
+```
+
+GKE Standard mode:
+
+```sh
+gcloud container clusters create onlineboutique \
+    --project=${PROJECT_ID} --zone=${ZONE} \
+    --workload-pool=${PROJECT_ID}.svc.id.goog \
+    --machine-type=n2-standard-2 --num-nodes=4
+```
+
+### 2. Enable Cloud Spanner on your project
+
+```sh
+gcloud services enable spanner.googleapis.com --project=${PROJECT_ID}
+```
+
+### 3. Create a Cloud Spanner instance, database, and table
+
+Note: See the documentation to list [available Spanner configuration names](https://cloud.google.com/spanner/docs/getting-started/set-up#run_the_gcloud_tool), or run `gcloud spanner instance-configs list --project=$PROJECT_ID`
+
+```sh
+SPANNER_REGION_CONFIG="<your-spanner-region-config-name>" # e.g. "regional-us-east5"
+SPANNER_INSTANCE_NAME=onlineboutique
+SPANNER_DATABASE_NAME=carts
+
+gcloud spanner instances create ${SPANNER_INSTANCE_NAME} \
+    --project=${PROJECT_ID} \
+    --description="online boutique backend" \
+    --config="${SPANNER_REGION_CONFIG}" \
+    --instance-type=free-instance
+
+gcloud spanner databases create ${SPANNER_DATABASE_NAME} \
+    --project=${PROJECT_ID} \
+    --instance="${SPANNER_INSTANCE_NAME}" \
+    --database-dialect=GOOGLE_STANDARD_SQL \
+    --ddl-file=./src/cartservice/ddl/CartItems.ddl
+```
+
+Note: If you see an error related to the `--instance-type` flag being `unrecognized`, run `gcloud components update`.
+
+### 4. Grant the default GCE/GKE service account read/write permission to Spanner
+
+The following works for configuring the default GCE/GKE and Kubernetes Service Accounts.
+It can be modified to use other service accounts by changing the values of the first three variables.
+
+```sh
+# Identify the service accounts, both on GCP and in Kubernetes
+GSA_NAME=$(gcloud iam service-accounts list --filter 'Compute Engine default' --format 'value(email)')
+NAMESPACE=default
+KSA_NAME=default
+
+# Grant the GCP service account databaseUser in Spanner
+gcloud projects add-iam-policy-binding $PROJECT_ID \
+    --member "serviceAccount:${GSA_NAME}" \
+    --role="roles/spanner.databaseUser"
+
+# Annotate the kubectl serviceAccountName with the GCP Service Account
+kubectl annotate serviceaccount ${KSA_NAME} \
+    --namespace=${NAMESPACE} \
+    iam.gke.io/gcp-service-account=${GSA_NAME}
+
+# Tell gcloud that the kubectl account maps to the GCP one
+gcloud iam service-accounts add-iam-policy-binding \
+    "${GSA_NAME}" --role roles/iam.workloadIdentityUser \
+    --member "serviceAccount:${PROJECT_ID}.svc.id.goog[${NAMESPACE}/${KSA_NAME}]"
+```
+
+### 5. Configure the Kubernetes manifest to use Spanner for the `cartservice`.
+
+```sh
+cp ./release/kubernetes-manifests.yaml ./release/updated-manifests.yaml
+sed -i "s/name: REDIS_ADDR/name: SPANNER_PROJECT/g" ./release/updated-manifests.yaml
+sed -i "s/value: \"redis-cart:6379\"/value: \"${PROJECT_ID}\"/g" ./release/updated-manifests.yaml
+sed -i "s/cartservice:v0.4.0/cartservice:v0.4.0-spanner/g" ./release/updated-manifests.yaml
```

Comment:
I add
```
sed -i "s/cartservice:v0.4.0/cartservice:v0.4.0-spanner/g" ./release/updated-manifests.yaml
```

I will be building and pushing a Spanner-supporting image at `gcr.io/google-samples/microservices-demo/cartservice:v0.4.0-spanner`.
Eventually, when we create the next release (e.g., `0.4.1`), that version's `cartservice` image will support Cloud Spanner by default (i.e., without the `-spanner` suffix).
So the above `sed` command I added is only relevant to anyone trying `spanner.md` with version `0.4.0`.

Reply:
✅  Just pushed at image to[ `gcr.io/google-samples/microservices-demo/cartservice:v0.4.0-spanner`](gcr.io/google-samples/microservices-demo/cartservice:v0.4.0-spanner).